capstone/src/main/java/com/google/step/clients/MetricsClient.java

```diff
@@ -0,0 +1,288 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.clients;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import java.util.*;
+
+/**
+ * This client is in charge of interacting with metricsServlet and the metrics entities.
+ */
+public class MetricsClient {
+
+    /**
+     * Empty constructor since we don't need any variables, we only use this class for the methods.
+     */
+    public MetricsClient() {}
+
+    private static DatastoreService dataStore = DatastoreServiceFactory.getDatastoreService();
+
+    /**
+     * Gets the current week's page views for one restaurant by specifying a restaurantKey.
+     * @param restaurantKey restaurant key of the restaurant we want to get views.
+     * @param restaurantName restaurant name in case no entity exists for this week's page views.
+     * @return returns a Map<String, Object> of the form:
```

Comment:
I still think this should return a defined type (a small class that has these fields as final member variables).

I get that it feels silly to do that because it just gets converted to JSON on the output anyway, but always think about code coupling and cohesion: Assume there will be OTHER pieces of code (not just servlets) that want to read these metrics and use this client. What's the right abstraction here.

I'd say, "returning something in json" is the servlet's job. Besides, if you return a basic data type, just about any JSON library has a way to serialize that to JSON. I'm sure this is true for the json org one you're using, and it's definitely true for Jackson (which I've used in the past).

Reply:
This has been changed to use two classes and it is a lot cleaner! It looks a lot better and feels easier to work with. Great advice!